### ISSUE TITLE
Disabling Windows Media Service Integration

### ISSUE THREAD

**Issue by TBXin** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#2100]
**OS:** Windows 10
**Issue Descriptions:** Hello there.
Just installed new version 4.1.1 and notice that Windows notifications behavior was changed.

When i press "Next/Prev song" button windows displays me notification like this one:

![](http://i.imgur.com/9hfNUAb.png)

Before this update notifications comes only from right bottom corner and they are looked like  small rectangle with a app icon and song name (also, i can turn off they at all).
I would appreciate if you add checkbox in the settings, which will be turn off new integration/behavior.

**Comment by MarshallOfSound** [comment]
@TBXin This behavior is not expected and I can't replicate it on any of my windows machines or VM's. 

* Is your windows up to date?
* Which "next/prev song button" are you pushing?
* What is your exact windows build number?

**Comment by TBXin** [comment]
@MarshallOfSound 

Windows 10 Enterprise, Version 1607 (OS Build 14393.693)

![](http://i.imgur.com/mjNSzJp.png)

**Comment by MarshallOfSound** [comment]
This is an identical build to me but I am on Windows 10 Pro not Enterprise (shouldn't affect it).

This is almost certainly a bug in windows itself or a temporary glitch.  Either a quick reboot will fix the issue or you will have to send feedback in the Windows 10 feedback app to inform Microsoft of this issue.

Regarding adding an option to disable this feature.  I will leave this issue open for the next 48 hours to determine if this is a wide-spread issue or if you are just unlucky and we'll go from there.

**Comment by TBXin** [comment]
@MarshallOfSound thanks for your fast response.

Seems to be this is really the Windows issue.

After rebooting this kind of notifications is gone.

**Comment by MarshallOfSound** [comment]
> After rebooting this kind of notifications are gone.

Good to hear 👍

**Comment by evonprime** [comment]
I'm having the same issue and a reboot unfortunately did not fix it. I also tried reinstalling GPMDP 4.1.1, but no luck. I'm on an up-to-date Windows 10 Home (version 1607, build 14393.693). 

The Windows media control popup only appears if I use the media hotkeys on my keyboard (Corsair K70). It does not appear if I use the media buttons in the main window or if I use the context menu of the icon in the Windows notification area.

This seems to be a separate notification from the desktop notification you can enable in the Labs settings, since it appears not matter if the Labs desktop notifications are enabled or disabled.

If it's a Windows issue, I guess you won't be able to do anything to fix it. I just wanted to let you know that there is more than one person having this issue.

**Comment by pcjl** [comment]
Also seeing this notification. Windows 10 Home build 14393.206. I'm using an XPS 13 (9350). While I'm fine with the new (albeit unintentional) notification, if I try skipping music with my media keys on my laptop's keyboard, it seems to skip twice (I'm assuming it skips with GPMDP like it used to, and then again with the new Windows Media Service Integration). I've tried rebooting but it had no effect. Furthermore, if I press my skip key twice (skipping four times), it seems to crash the program and no longer responds until I forcefully quit the program altogether.

**Comment by evonprime** [comment]
I can confirm that hitting next/previous on my keyboard media hotkeys skips over a song in the list, I didn't notice at first. Also, for some reason the very first time I hit play/pause on my keyboard, the Windows media GUI shows up but the song doesn't pause. Hitting it again then pauses and after that, it works every time as intended. I can also reproduce the program freezing if I hit next a few times in a row.

**Comment by MarshallOfSound** [comment]
I'm thinking as a solution to the "double hit" issue I can detect to commands in a short interval, and if it happens 3 times in a row pop up a warning that allows users to disable the global media hotkeys but retain the windows service keys.  Will look into it more at some point

**Comment by TBXin** [comment]
Strange things...
Today this notifications are back.

I can assume that they are back, when i clicked the media key (media next) when Windows Volume Level Popup was on screen.

**Comment by TBXin** [comment]
Also i confirming, that Media Next/Prev keys skipping 2 songs at once.

**Comment by strigona-worksight** [comment]
I'm experiencing this issue when using the Prev/Next buttons on my Logitech keyboard (K520).

However I'm **NOT** experiencing it when using the Prev/Next buttons on my Logitech G930 (I know there was previously compatibility issues here).

**Comment by mstrelan** [comment]
Same issue here. I don't mind the notification except that it's positioned in the top left of my screen, offset just enough to be entirely in the way. If anyone figures out how to move it to the bottom right please let me know.

**Comment by kdenney** [comment]
@MarshallOfSound 

Can this issue be reopened? Something definitely changed in version 4.1.1 that cause this popup to appear because it never happened before then. Many people (myself included) are still having the issue even after a reboot. 

Also, I've found threads elsewhere where spotify added this "feature" and had to add an option to their preferences pane to disable it. ([link](https://community.spotify.com/t5/Desktop-Linux-Windows-Web-Player/Windows-10-Left-corner-song-banner-pause-stop-FIX-IT/td-p/1480262))

Looking at the changes in 4.1.1, maybe it was somehow introduced with this change?
_NEW: Windows Media Service Integration (Lock screen music controls)_

**Comment by MarshallOfSound** [comment]
@kdenny No :P

I've already resorting to the same solution that Spotify did and added an option to disable the W10 Media Service integration.

And FWIW it's supposed to pop-up when you use change your volume :D

**Comment by TBXin** [comment]
>added an option to disable the W10 Media Service integration.

Nice!

**Comment by kdenney** [comment]
@MarshallOfSound  Oh great! Thanks for the update; hopefully disabling that option will make the popup go away forever. :P

Do you know when this change will be pushed to a live version?

**Comment by MarshallOfSound** [comment]
When I have time to make a release :D. It's a non-trivial process, especially now I need to build Linux releases on VM's.

**Comment by kdenney** [comment]
Ah, gotcha. Well thanks so much for the effort! I forgot to mention it earlier: I absolutely love this app. :D

**Comment by dramamoose** [comment]
Is there a way to configure the program choose one over the other? Personally, I like the Windows 10 music controls, but I'd like to prevent the double skipping issue by disabling global hotkeys within the app.

**Comment by IndefiniteBen** [comment]
I've been frustrated by this issue for a while and am just now looking into it... Most of the duplicate issues direct people to this issue or #2107; neither of which actually explain/cover why it's been closed despite not being fixed, i.e., using Win10 Media Service still results in double skipping in 4.3.0. The current solution is to turn off the Win10 media service that is causing the double-skip issue.
It is mentioned in #2253 what the [underlying cause](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/2253#issuecomment-284340863) of this issue is, with some discussion ending on 27 Mar. The current open issue with discussion on this is #2364. I hope this helps anyone confused by the issues stating it's fixed despite the fact it still happens.

**tl;dr**
Turning off Win10 media service is the current solution for stopping double skips; no-compromise-solution discussion: #2364
